UnityProjects/MRTKDevTemplate/Assets/Scripts/SpeechRecognitionHandler.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Subsystems;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Microsoft.MixedReality.Toolkit.Examples.Demos
+{
+    /// <summary>
+    /// Demonstration script showing how to subscribe to and handle
+    /// events fired by SpeechRecognitionSubsystem.
+    /// </summary>
+    [AddComponentMenu("MRTK/Examples/Speech Recognition Handler")]
+    public class SpeechRecognitionHandler : MonoBehaviour
+    {
+
+        /// <summary>
+        /// Wrapper of UnityEvent&lt;string&gt; for serialization.
+        /// </summary>
+        [System.Serializable]
+        public class StringUnityEvent : UnityEvent<string> { }
+
+        /// <summary>
+        /// Event raised while the user is talking. As the recognizer listens, it provides text of what it's heard so far.
+        /// </summary>
+        public StringUnityEvent OnSpeechRecognizing;
```

Comment:
Can these be private serialized fields?

Reply:
They could, but I was thinking about the possibility of users accessing them via code. I know I can do private serialized fields + properties, but that seemed a bit overkill for a sample script to me. Open to other thoughts though!

Reply:
I've always tried to maintain the same code quality across MRTK code and sample code, for consistency and familiarity. All of this code may end up in a dev's app in one way or another!